Source/Wanderer/Private/Character/WandererPlayer.cpp

```diff
@@ -13,3 +15,39 @@ AWandererPlayer::AWandererPlayer() {
   PlayerCamera->SetupAttachment(SpringArm);
 
 }
+
+
+
+void AWandererPlayer::BeginPlay()
+{
+	Super::BeginPlay();
+	// Registering the input mapping context with the input subsystem
+	if (APlayerController* PlayerController = Cast<APlayerController>(GetController())) {
+		if (UEnhancedInputLocalPlayerSubsystem* Subsystem = ULocalPlayer::GetSubsystem<UEnhancedInputLocalPlayerSubsystem>(PlayerController->GetLocalPlayer())) {
+			Subsystem->AddMappingContext(InputMappingContext, 0);
+		}
+	}
+}
+
+void AWandererPlayer::SetupPlayerInputComponent(UInputComponent* PlayerInputComponent) {
+	Super::SetupPlayerInputComponent(PlayerInputComponent);
+
+	// Cast to use enhanced input features
+	UEnhancedInputComponent* EnhancedInputComponent = CastChecked<UEnhancedInputComponent>(PlayerInputComponent);
+	if (EnhancedInputComponent) {
+		EnhancedInputComponent->BindAction(MoveAction, ETriggerEvent::Triggered, this, &AWandererPlayer::OnMove);
+	}
+}
+
+// Handle movement input 
+void AWandererPlayer::OnMove(const FInputActionValue& Value) {
+	// use 2D vector (from Kadyn and vid)
+	const FVector2D MovementInput = Value.Get<FVector2D>();
+
+	//UE_LOG(LogTemp, Warning, TEXT("Move Input: %s"), *MovementInput.ToString());

```

Comment:
We can remove this debug string altogether instead of commenting, this is something we only need once and if we need it again, we can just retype it.